### ISSUE TITLE
How to remove a requirement?

### ISSUE THREAD

**Issue by nicolasfranck** [perl-carton/carton#277]
If I remove a requirement from `cpanfile`, and run `carton install` again,
the requirement is not removed from the snapshot.

If I remove the requirement manually from the snapshot and cpanfile, and run `carton install` again,
the requirement  is added again to the snapshot. It is like the dependency tree
is derived from the directory tree, and not from the snapshot.

If I remove the requirement from the cpanfile, the snapshot and from the filesystem,
only then it stays out of the snapshot ..

Any idea how to make this more manageable? Because like this old dependencies
will remain forever..

**Comment by bbrtj** [comment]
If anyone is looking for a solution - seems like deleting dependency from `cpanfile`, module directory from `local/lib/perl5/envname` (`local/lib/perl5/amd64-freebsd` on my machine) and then running `carton install` is the way to remove something from the cpanfile.snapshot. Sadly, think its dependencies are not removed, so you have to snipe those manually.

**Comment by miyagawa** [comment]
Yes, that's correct, Carton doesn't have a clean way to remove a module, since that state is not managed by Carton and is preserved in `local` - You have to blow away `local` (to get everything installed from scratch, which takes time) or manually remove files from `local` (tedious, but could save time).

plug: Carmel has a much better way of doing updates like this and I'm actively working on it.

**Comment by jonasbn** [comment]
Carmel can be found at: https://github.com/miyagawa/carmel
